server/server.js

```diff
@@ -138,4 +137,92 @@ app.get('/sidebar/all', (req, res) => {
   });
 });
 
+// Create
+app.post('/sidebar/all', async (req, res) => {
+  console.log('POST request to /sidebar/all');
+  const newDocument = req.body;
+  const newCourseId = newDocument.courseId;
+  console.log(newDocument);
+  // change string date to date type
+  newDocument.price.saleEndDate = new Date(newDocument.price.saleEndDate);
+
+  if (typeof newCourseId !== 'number') {
+    res.status(400).send('Sorry, invalid request: courseId is not a number');
+  } else {
+    db.getSidebar({ courseId: newCourseId }, async (err, docs) => {
+      console.log('return from courseId query:', docs);
+      if (err) {
+        console.warn('Error Occured :', err);
+      } else if (docs[0] === undefined) {
+        console.log('courseId is available!');
+        await db.postAll(newDocument)
+          .then((result) => {
+            console.log(result);
+            res.status(201).send(result);
+          })
+          .catch((error) => {
+            console.warn('Error occured during create (server side): ', error);
+          });
+      } else {
+        res.status(400).send('Sorry, courseId already exists.');
+      }
+    });
+  }
+});
+
+// Delete
+app.delete('/sidebar/all', async (req, res) => {
+  console.log('DELETE request to /sidebar/all', req.query);
+  const courseId = req.query;
+  await db.deleteAll(courseId)
+    .then((result) => {
+      if (result) {
+        res.status(204);
+      } else {
+        throw Error('Check DB side');
+      }
+      res.end();
+    })
+    .catch((error) => {
+      console.warn('Error occured during delete', error);
+      res.status(400).send('Sorry, Error in deleting occured.');
+      res.end();
+    });
+});
+
+// Update
+app.put('/sidebar/all', async (req, res) => {
+  const updateDoc = req.body;
+  const { courseId } = updateDoc;
+  console.log('PUT request to /sidebar/all { courseId: ', courseId, ' }');
+  const updating = [];
+  if (updateDoc.price !== undefined) {
+    updating.push('price');
+  }
+  if (updateDoc.sidebar !== undefined) {
+    updating.push('sidebar');
+  }
+  if (updateDoc.previewVideo !== undefined) {
+    updating.push('previewVideo');
+  }
+
+  if (typeof courseId !== 'number') {
+    res.status(400).send('Sorry, invalid request: courseId is not a number');
```

Comment:
I think you are free to assume courseId is a valid request in this case. But it's great that you take that situation into consideration.

Reply:
thanks!